src/test/java/org/jvnet/hudson/test/JenkinsRuleTest.java

```diff
@@ -424,6 +429,31 @@ public void waitForCompletion() throws Exception {
         j.assertBuildStatusSuccess(j.waitForCompletion(b));
     }
 
+    @Test
+    public void restart() throws Throwable {
+        // preserve relevant properties
+        URL previousUrl = j.getURL();
+        Description previousTestDescription = j.testDescription;
+        File previousRoot = j.jenkins.getRootDir();
+
+        // create some configuration
+        j.createFreeStyleProject();
+        assertThat(j.jenkins.getJobNames(), hasSize(1));
+
+        // restart the instance with same port and new JENKINS_HOME
+        j.restart();
```

Comment:
I'm not sure how this tests that restart is actually working?

Does the port stay the same across restarts?

If it doesn't then you could test that the port changes, similar to this test:
https://github.com/jenkinsci/jenkins-test-harness/blob/cb93195cbbfd9ea8802692c0d9f92d31543b6afc/src/test/java/org/jvnet/hudson/test/RestartableJenkinsRuleTest.java#L42-L45

Reply:
Yes, the port will stay the same.
The port is implicitly tested by comparing the URL of the instances. Should I rather only check the port?

Reply:
ah I see